hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -431,7 +432,9 @@ protected Expression getFilterPredicates(Configuration configuration) {
     // Get the available CG datamaps and prune further.
     DataMapExprWrapper cgDataMapExprWrapper = DataMapChooser.get()
         .chooseCGDataMap(getOrCreateCarbonTable(job.getConfiguration()), resolver);
-    if (cgDataMapExprWrapper != null) {
+    if (cgDataMapExprWrapper != null &&
+        DataMapStatusManager.isDataMapEnabled(
+            cgDataMapExprWrapper.getDataMapSchema().getDataMapName())) {
```

Comment:
This logic should be moved to DataMapStoreManager and DataMapChooser. There when it gets all datamaps it should only enabled datamaps. There should not be any changes here.

Reply:
logic modified, please check again